content/kubeone/master/guides/mirror_registries/_index.en.md

```diff
@@ -0,0 +1,103 @@
++++
+title = "Using Mirror Registries"
+date = 2021-12-01T12:00:00+02:00
+enableToc = true
++++
+
+## Introduction
+
+{{% notice warning %}}
+The ContainerRegistry API is available only starting with KubeOne 1.4 and
+newer. Additionally, Docker supports only configuring the mirror registry for
+`docker.io` images. As an alternative, you can follow the
+[Overwriting Image Registries guide]({{< ref "../registry_configuration" >}})
+guide, however it's considered as legacy. We recommend upgrading to KubeOne 1.4
+and migrating to containerd.
+{{% /notice %}}
+
+This guide describes how to use mirror registries for images deployed by
+KubeOne (Kubernetes core components, CNI plugins...). This is useful if don't
+have access to the original registries (e.g. you're having an offline setup)
+or if you want to workaround Docker Hub pull limits. To accomplish this, this
+guide uses the [ContainerRegistry API][containerreg-api].
+
+## Prerequisites
+
+This guide assumes that:
+
+* you have an image registry up and running
+* all your nodes in the cluster can access the image registry
+
+If you don't have an image registry, you can check out the
+[Docker Registry][docker-reg-guide] as a possible solution.
+
+## Configuring Mirror Registries
+
+This section describes how to configure the mirror registries.
+
+### Docker
+
+{{% notice warning %}}
+Docker supports configuring mirror registries only for `docker.io` images.
+{{% /notice %}}
+
+You can configure the mirror registries by adding the `containerRuntime`
+stanza to your KubeOne configuration file, such as:
+
+```yaml
+apiVersion: kubeone.k8c.io/v1beta2
+kind: KubeOneCluster
+versions:
+  kubernetes: 1.22.5
+cloudProvider:
+  aws: {}
+containerRuntime:
+  docker:
+    registryMirrors:
+      - http://host1.tld
+      - https://host2.tld
+```
+
+For more information about the ContainerRuntime API for Docker, see the
+[API reference][containerruntime-docker].
+
+With this done, you can reconcile your cluster by running `kubeone apply`.
+
+### Containerd
+
+You can configure the mirror registries by adding the `containerRuntime`
+stanza to your KubeOne configuration file, such as:
+
+```yaml
+apiVersion: kubeone.k8c.io/v1beta2
+kind: KubeOneCluster
+versions:
+  kubernetes: 1.22.5
+cloudProvider:
+  aws: {}
+containerRuntime:
+  containerd:
+    registries:
+      myunknown.tld:
+        mirrors:
+        - host1.tld
+        - https://host2.tld
+        tlsConfig:
+          insecureSkipVerify: true
+        auth:
+          username: "user1"
+          password: "insecure"
```

Comment:
WDYT about mentioning the fact that registry auth can be in dedicated credentials file?